### ISSUE TITLE
Add `isinstance` in `if` condition check as a special case

### ISSUE THREAD

**Issue by a4abhishek** [PyCQA/pylint#1990]
### Steps to reproduce
1. Write this simple code and save it:
```py
import subprocess

try:
    subprocess.check_call(['ls', '-']) # Deliberately made error in this line
except Exception as err:
    if isinstance(err, subprocess.CalledProcessError):
        print 'Subprocess error occured. Return code:', err.returncode
        # Error in above line.......................here^^^
    else:
        print 'An error occured:', str(err)
```
2. Run `pylint` on this file.
3. Output of Pylint:
```
$ pylint pylint_test.py 
No config file found, using default configuration
************* Module pylint_test
C:  1, 0: Missing module docstring (missing-docstring)
W:  5, 7: Catching too general exception Exception (broad-except)
E:  7,56: Instance of 'Exception' has no 'returncode' member (no-member)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 2.86/10, -2.86)
```

### Current behavior
I is showing Error in line: 7 which should not be an error.

### Expected behavior
It should not show Error in line-7 as I am already checking if `err` is an instance of `subprocess.CalledProcessError` which has member `returncode`.

### pylint --version output
```
No config file found, using default configuration
pylint 1.8.3, 
astroid 1.6.2
Python 2.7.12 (default, Dec  4 2017, 14:50:18) 
[GCC 5.4.0 20160609]
```

**Comment by PCManticore** [comment]
Thanks for creating this issue @a4abhishek ! This is a well known problem to us, we simply don't support any control flow understanding whatsoever right now, basically we disregard any `if` guard and infer the potential values of a variable from all the branches, which is the reason why you're also seeing this error. There is no progress yet, but we have dozens of issues that will be solved once we'll add this inference support.

**Comment by a4abhishek** [comment]
Thanks @PCManticore for considering this.
